# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.h
# Commit: 68219c132efc
# Full Hash: 68219c132efce82c72e004ab3651f7472e63a740
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2019-12-12 21:50:40
# Regressor Bug: 1599043
# File Overlap Count: 1
# Description:
#   Bug 1599043 - Part 1: Add a ContentBlockingLog into the WindowGlobalParent r=dimi,Ehsan
#   
#   This adds a ContentBlockingLog into the WindowGlobalParent. This
#   ContentBlockingLog is bascially a copy of the log in the content
#   process. This log in parent is needed for the OnContentBlockingEvent in
# ==============================================================================

diff -r 75643c678e24 -r 68219c132efc dom/ipc/WindowGlobalParent.h
--- a/dom/ipc/WindowGlobalParent.h	Thu Dec 12 16:38:54 2019 +0200
+++ b/dom/ipc/WindowGlobalParent.h	Wed Dec 11 15:00:20 2019 +0000
@@ -7,6 +7,7 @@
 #ifndef mozilla_dom_WindowGlobalParent_h
 #define mozilla_dom_WindowGlobalParent_h
 
+#include "mozilla/AntiTrackingCommon.h"
 #include "mozilla/RefPtr.h"
 #include "mozilla/dom/DOMRect.h"
 #include "mozilla/dom/PWindowGlobalParent.h"
@@ -16,6 +17,7 @@
 #include "nsISupports.h"
 #include "mozilla/dom/WindowGlobalActor.h"
 #include "mozilla/dom/CanonicalBrowsingContext.h"
+#include "mozilla/dom/ContentBlockingLog.h"
 
 class nsIPrincipal;
 class nsIURI;
@@ -130,6 +132,14 @@
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
 
+  void NotifyContentBlockingEvent(
+      uint32_t aEvent, nsIRequest* aRequest, bool aBlocked, nsIURI* aURIHint,
+      const nsTArray<nsCString>& aTrackingFullHashes,
+      const Maybe<AntiTrackingCommon::StorageAccessGrantedReason>& aReason =
+          Nothing());
+
+  ContentBlockingLog* GetContentBlockingLog() { return &mContentBlockingLog; }
+
  protected:
   const nsAString& GetRemoteType() override;
   JSWindowActor::Type GetSide() override { return JSWindowActor::Type::Parent; }
@@ -178,6 +188,8 @@
 
   // True if this window has a "beforeunload" event listener.
   bool mHasBeforeUnload;
+
+  ContentBlockingLog mContentBlockingLog;
 };
 
 }  // namespace dom
